Lib/http/server.py

```diff
@@ -856,7 +881,11 @@ def copyfile(self, source, outputfile):
         to copy binary data as well.
 
         """
-        shutil.copyfileobj(source, outputfile)
+        if start_byte is not None and end_byte is not None:
+            source.seek(start_byte)
+            outputfile.write(source.read(end_byte))
```

Comment:
Fatal error. When you use `2-5`, the start_byte is 2 and end_byte is 5. Then you `read(5)`.
The correct behavior is `read(4)`, which is `end_byte-start_byte+1`. Besides Content-Length needs changing, too.